openedx/features/content_type_gating/partitions.py

```diff
@@ -0,0 +1,220 @@
+                font: normal normal 400 normal 16px / 25.6px 'Open Sans';
+            }
+
+            .content-paywall img {
+                height: 60px;
+            }
+        """))
+        return frag
+
+
+class ContentTypeGatingPartitionScheme(object):
+    """
+    This scheme implements the Content Type Gating permission partitioning.
+
+    This partitioning is roughly the same as the verified/audit split, but also allows for individual
+    schools or courses to specify particular learner subsets by email that are allowed to access
+    the gated content despite not being verified users.
+    """
+
+    LIMITED_ACCESS = Group(settings.CONTENT_TYPE_GATE_GROUP_IDS['limited_access'], 'Limited-access Users')
+    FULL_ACCESS = Group(settings.CONTENT_TYPE_GATE_GROUP_IDS['full_access'], 'Full-access Users')
+
+    @classmethod
+    def get_group_for_user(cls, course_key, user, user_partition, **kwargs):
+        """
+        Returns the Group for the specified user.
+        """
+
+        # First, check if we have to deal with masquerading.
+        # If the current user is masquerading as a specific student, use the
+        # same logic as normal to return that student's group. If the current
+        # user is masquerading as a generic student in a specific group, then
+        # return that group.
+        if get_course_masquerade(user, course_key) and not is_masquerading_as_specific_student(user, course_key):
+            return get_masquerading_user_group(course_key, user, user_partition)
+
+        # For now, treat everyone as a Full-access user, until we have the rest of the
+        # feature gating logic in place.
+        return cls.FULL_ACCESS
+
+    @classmethod
+    def create_user_partition(cls, id, name, description, groups=None, parameters=None, active=True):  # pylint: disable=redefined-builtin, invalid-name, unused-argument
+        """
+        Create a custom UserPartition to support dynamic groups.
+
+        A Partition has an id, name, scheme, description, parameters, and a list
+        of groups. The id is intended to be unique within the context where these
+        are used. (e.g., for partitions of users within a course, the ids should
+        be unique per-course). The scheme is used to assign users into groups.
+        The parameters field is used to save extra parameters e.g., location of
+        the course ID for this partition scheme.
+
+        Partitions can be marked as inactive by setting the "active" flag to False.
+        Any group access rule referencing inactive partitions will be ignored
+        when performing access checks.
+        """
+        return ContentTypeGatingPartition(
+            id,
+            unicode(name),
+            unicode(description),
+            [
+                cls.LIMITED_ACCESS,
+                cls.FULL_ACCESS,
+            ],
+            cls,
+            parameters,
+            # N.B. This forces Content Type Gating to always be active on every course
```

Comment:
could you clarify the comment? I'm assuming this is just for the partition
because we don't want content gating on in every course?